### PR TITLE
FEAT: make correlation IDs consumer configurable

### LINKED ISSUE #102: Make correlation operation ID configurable for the consumer
**Is your feature request related to a problem? Please describe.**
The `CorrelationInfo` will generated a guid-like `OperationId` when there's no `.TraceIdentifier` found for the received requests.

**Describe the solution you'd like**
The consumer should be able to configure the way we generate the `OperationId`, including (maybe) to bypass the use of the `.TraceIdentifier`.

**Additional context**
Initial issue: #67 
Initial correlation PR: #92

### LINKED ISSUE #103: Make correlation info transaction ID configurable for the consumer
**Is your feature request related to a problem? Please describe.**
The `CorrelationInfo` will generate a guid-like `Transactionid` when there are no existing transaciton ID's in the request (and we have specified to generate when when not defined).

**Describe the solution you'd like**
The consumer should be able to configure the way we generate the `TransactionId`.

**Additional context**
Initial issue: #67
Initial correlation PR: #92

### PR DESCRIPTION
**Pull request by stijnmoreels**
Provide functions in the options to custommize the generation of the correlation ID's.

Closes #102 
Closes #103

### CODE REVIEW

**Comment by stijnmoreels** [comment]
> LGTM, only thing missing are docs - Can you add them please?

Yeah, sure.
You're OK, with the added options being functions and not object instances?

**Comment by tomkerkhove** [comment]
:+1:

File: docs/features/correlation.md [hunk 1]
```diff
@@ -72,6 +72,10 @@ public void ConfigureServices(IServiceCollection services)
         // The header to look for in the HTTP request, and will be set in the HTTP response (default: X-Transaction-ID).
         options.Transaction.HeaderName = "X-Transaction-ID";
 
+        // The function that will generate the transaction ID, when the `.GenerateWhenNotSpecified` is set to `false` and the request doesn't contain the header.
+        // (default: new `Guid`).
+        options.Transaction.GenerateId = () => Guid.NewGuid().ToString();
```
Comment (tomkerkhove) [inline docs/features/correlation.md]: Maybe use something that differentiates from default such as :
```suggestion
        options.Transaction.GenerateId = () => $"Transaction-{Guid.NewGuid().ToString()}";
```

Comment (stijnmoreels) [reply docs/features/correlation.md]: Sure!

**Comment by stijnmoreels** [comment]
Done?

**Pull request merged**